### PR TITLE
Adding section on Accessibility Levels and Rules

### PR DESCRIPTION
**Pull request by gusthoff**

### CODE REVIEW

File: content/courses/advanced-ada/chapters/access_types.rst [hunk 1]
```diff
@@ -1577,17 +1577,732 @@ type) and display its information by calling the :ada:`Show` procedure.
...
 
-    Complete section!
+First, let's talk a bit about
+`lifetime of objects <https://en.wikipedia.org/wiki/Variable_(computer_science)#Scope_and_extent>`_.
+We assume you understand the concept, so this section is very short.
+
+In very simple terms, the lifetime of an object indicates when an object still
+has relevant information. For example, if a variable :ada:`V` gets out of
+scope, we say that its lifetime has ended. From this moment on, :ada:`V`
+doesn't have any relevant information anymore.
```
Comment (pat-rogers) [inline content/courses/advanced-ada/chapters/access_types.rst:1607]: It seems better IMO to say that the object no longer exists.

File: content/courses/advanced-ada/chapters/access_types.rst [hunk 2]
```diff
@@ -1577,17 +1577,732 @@ type) and display its information by calling the :ada:`Show` procedure.
...
+reference. Later on, we'll discuss how dangling references may occur using
+:ref:`unchecked deallocation <Adv_Ada_Unchecked_Deallocation_Dangling_References>`.
+
+Dangling references are created when we have an access value pointing to an
+object whose lifetime has ended, so it becomes a  non-existent object. This
+could occur, for example, when an access value still points to an object
+:ada:`X` that has gone out of scope.
+
+As mentioned in the previous section, the accessibility rules of the Ada
+language prevent that such situations ever happen! In fact, whenever possible,
```
Comment (pat-rogers) [inline content/courses/advanced-ada/chapters/access_types.rst:1939]: The phrase "language prevent that such situations ever happen!" is a little awkward in English. Maybe something like this:  "language ensure that such situations never happen!"

File: content/courses/advanced-ada/chapters/access_types.rst [hunk 3]
```diff
@@ -1577,17 +1577,732 @@ type) and display its information by calling the :ada:`Show` procedure.
...
+          --             I_Var_2 has ended!
+
+Here, we're introducing the :ada:`A1 := A2` assignment. The problem with this
+is that :ada:`I_Var_2`\ 's lifetime ends when the :ada:`Inner_Block` finishes,
+but :ada:`A1` would continue to refer to an :ada:`I_Var_2` object that doesn't
+exist anymore |mdash| thereby creating a dangling reference.
+
+Even though we're actually not assigning :ada:`A2` to :ada:`A1` in the original
+code, we could have done it. The accessibility rules prevent that such an error
+is ever introduced to the program.
```
Comment (pat-rogers) [inline content/courses/advanced-ada/chapters/access_types.rst:2049]: Same issue here. Suggest: "The accessibility rules ensure that such an error is never introduced into the program."

File: content/courses/advanced-ada/chapters/access_types.rst (hunk 2 above)
Comment (gusthoff) [reply content/courses/advanced-ada/chapters/access_types.rst:1939]: Thanks! As a non-native speaker, I'm always grateful for this kind of corrections. :-)

**Review by pat-rogers — APPROVED** [review]

**Pull request merged**